ingestion-edge/contract-tests/test_api.py

```diff
@@ -0,0 +1,110 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import datetime
+from dateutil.parser import parse
+from typing import List
+import requests
+
+
+def test_dockerflow(requests_session: requests.Session, server: str):
+    r = requests_session.get(server + "/__heartbeat__")
+    r.raise_for_status()
+    assert r.json() == {"checks": {}, "details": {}, "status": "ok"}
+
+    r = requests_session.get(server + "/__lbheartbeat__")
+    r.raise_for_status()
+    assert r.text == ""
+
+    r = requests_session.get(server + "/__version__")
+    r.raise_for_status()
+    assert sorted(r.json().keys()) == ["build", "commit", "source", "version"]
+
+
+def test_publish(
+    create_pubsub_resources: bool,
+    requests_session: requests.Session,
+    server: str,
+):
+    from ingestion_edge.config import ROUTE_TABLE
+    from ingestion_edge.publish import client as publisher
+    from google.cloud.pubsub import SubscriberClient
```

Comment:
Imports in a function are usually a think I justifiy with a comment, and only do if it's required.